### PR TITLE
common: blkdev cleanup, sanity checks, etc

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/common/blkdev.cc [hunk 1]
```diff
@@ -9,29 +11,26 @@
...
 
 static const char *sandbox_dir = "";
 
 void set_block_device_sandbox_dir(const char *dir)
 {
-  if (dir)
-    sandbox_dir = dir;
-  else
-    sandbox_dir = "";
+  dir ? sandbox_dir = dir : sandbox_dir = "";
```
Comment (wjwithagen) [inline src/common/blkdev.cc]: @shinobu-x 
Don't really like the "hackish" way of trying to do the same as the it-then-else.
Would just leave it as is.

File: src/common/blkdev.cc [hunk 2]
```diff
@@ -67,15 +66,12 @@ int get_block_device_base(const char *dev, char *out, size_t out_len)
...
+    (!realpath(dev, realname) || (strncmp(realname, "/dev/", 5) != 0)))
       return -EINVAL;
-    }
-  }
 
-  if (strlen(realname))
-    strncpy(devname, realname + 5, PATH_MAX - 5);
-  else
+  strlen(realname) ?
+    strncpy(devname, realname + 5, PATH_MAX - 5) :
```
Comment (wjwithagen) [inline src/common/blkdev.cc]: @shinobu-x 
Same here

File: src/common/blkdev.cc [hunk 3]
```diff
@@ -204,62 +204,22 @@ int block_device_model(const char *devname, char *model, size_t max)
...
   struct stat st;
-  int r = fstat(fd, &st);
-  if (r < 0) {
+  if (fstat(fd, &st) < 0) {
     return -EINVAL;  // hrm.
   }
   dev_t devid = S_ISBLK(st.st_mode) ? st.st_rdev : st.st_dev;
   char *t = blkid_devno_to_devname(devid);
   if (!t) {
+    free(t);
```
Comment (wjwithagen) [inline src/common/blkdev.cc]: @shinobu-x 
t == NULL, so why try to free it?

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Pull request closed without merging**